etc/calc_release_version.py

```diff
@@ -190,7 +203,7 @@ def process_and_sort_tags(tags):
         tag_parts = tag.split('-')
         if len(tag_parts) >= 2 and tag_parts[0] not in processed_and_sorted_tags:
             processed_and_sorted_tags.append(tag)
-    processed_and_sorted_tags.sort(key=LooseVersion)
+    processed_and_sorted_tags.sort(key=Version)
```

Comment:
Tasks appear to fail on this line:

https://spruce.mongodb.com/task/cxx_driver_debian10_release_latest_compile_and_test_with_static_libs_extra_alignment_patch_ed5a02fb0bf6135e25c45723ba5d768d59cb785e_646f9256d6d80a32b136916f_23_05_25_16_52_39/logs?execution=0

Reply:
The failure seems to be a result of Evergreen executing the build while checked out to branch other than `master` (namely, the branch from the PR or `cxx-2694`). I'm working out a fix.